File: composeApp/src/androidMain/kotlin/org/example/project/presentations/screen/login/components/AuthenticationErrorBottomSheet.kt
package org.example.project.presentations.screen.login.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import org.example.project.presentations.components.ButtonView
import org.example.project.presentations.theme.LocalExtendedColors

@OptIn(ExperimentalMaterial3Api::class)
@Preview
@Composable
fun AuthenticationErrorBottomSheet(
    modifier: Modifier = Modifier,
    onRetry: () -> Unit = {}
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .background(Color.White)
            .padding(horizontal = 20.dp, vertical = 16.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {

        Box(
            modifier = Modifier
                .width(40.dp)
                .height(4.dp)
                .background(Color.LightGray, RoundedCornerShape(10.dp))
        )

        Spacer(modifier = Modifier.height(20.dp))

        Column(
            modifier = Modifier
                .fillMaxWidth()
                .background(
                    color = Color(0xFFFBE9E9),
                    shape = RoundedCornerShape(16.dp)
                )
                .padding(20.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {

            Box(
                modifier = Modifier
                    .size(56.dp)
                    .background(
                        Color(0xFFFFD6D6),
                        CircleShape
                    ),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = Icons.Default.Warning,
                    contentDescription = null,
                    tint = LocalExtendedColors.current.red,
                    modifier = Modifier.size(28.dp)
                )
            }

            Spacer(modifier = Modifier.height(16.dp))

            Text(
                text = "Authentication Error !",
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.Bold,
                color = LocalExtendedColors.current.mainRed
            )

            Spacer(modifier = Modifier.height(10.dp))

            Text(
                text = "We couldn't find an account matching those credentials. Please try again !",
                style = MaterialTheme.typography.bodyMedium,
                color = LocalExtendedColors.current.mainRed,
                textAlign = TextAlign.Center
            )
        }

        Spacer(modifier = Modifier.height(24.dp))

        ButtonView(
            text = "Retry",
            backgroundColorRes = LocalExtendedColors.current.red,
            textColorRes = Color.White,
            enabled = true,
            onClick = onRetry
        )
    }
}

File: iosApp/iosApp/Presentation/Components/AuthenticationErrorBottomSheet.swift
//
//  AuthenticationErrorBottomSheet.swift
//  iosApp
//
//  Created by Tuan Nguyen on 3/27/26.
//

import SwiftUI

struct AuthenticationErrorBottomSheet: View {
    var onRetry: () -> Void = {}

    var body: some View {
        VStack(alignment: .center, spacing: 0) {
            
            Spacer().frame(height: 20)
            
            VStack(spacing: 0) {
                
                ZStack {
                    Circle()
                        .fill(Color(red: 1.0, green: 0.84, blue: 0.84))
                        .frame(width: 56, height: 56)
                    
                    Image(systemName: "exclamationmark.triangle.fill")
                        .resizable()
                        .scaledToFit()
                        .frame(width: 28, height: 28)
                        .foregroundColor(.red)
                }
                
                Spacer().frame(height: 16)
                
                Text("Authentication Error !")
                    .font(.system(size: 16, weight: .bold))
                    .foregroundColor(.red) // mainRed
                
                Spacer().frame(height: 10)
                
                Text("We couldn't find an account matching those credentials. Please try again !")
                    .font(.system(size: 14))
                    .foregroundColor(.red)
                    .multilineTextAlignment(.center)
            }
            .frame(maxWidth: .infinity)
            .padding(20)
            .background(Color(red: 0.98, green: 0.91, blue: 0.91))
            .cornerRadius(24)
            
            Spacer().frame(height: 24)
            
            ButtonView(
                text: "Retry",
                textSize: 16,
                textColor: .white,
                backgroundColor: .red,
                enabled: true,
                onClick: onRetry
            )
        }
        .padding(.horizontal, 36)
        .padding(.top, 16)
        .frame(maxWidth: .infinity)
        .background(Color.white)
    }
}
